modules/caddyhttp/fileserver/staticfiles.go

```diff
@@ -330,40 +360,50 @@ func sanitizedPathJoin(root, reqPath string) string {
 	// if the length is 1, then it's a path to the root,
 	// and that should return ".", so we don't append the separator.
 	if strings.HasSuffix(reqPath, "/") && len(reqPath) > 1 {
-		path += string(filepath.Separator)
+		path += separator
 	}
 
 	return path
 }
 
-// fileHidden returns true if filename is hidden
-// according to the hide list.
+// fileHidden returns true if filename is hidden according to the hide list.
+// filename must be a relative or absolute file system path, not a request
+// URI path. It is expected that all the paths in the hide list are absolute
+// paths or are singular filenames (without a path separator).
 func fileHidden(filename string, hide []string) bool {
-	sep := string(filepath.Separator)
+	if len(hide) == 0 {
+		return false
```

Comment:
Does it really save anything to do this condition? It'll just skip past the loop and return false if it's empty anyways.

Reply:
Saves a possible system call on the next line :point_down:

Reply:
I guess.